sycl/doc/extensions/experimental/sycl_ext_intel_fpga_task_sequence.asciidoc

```diff
@@ -0,0 +1,498 @@
+ structure _D_, then it does so and some other task thread in _S_ is
+ strengthened to have concurrent forward progress guarantee. If a task thread
+ cannot write its results to _D_, the task thread blocks until space is
+ available.
+
+The two examples below, respectively, show the following:
+
+1. How strengthening of a work item strengthens the task threads.
+
+2. How a task thread delegates its progress guarantee to other task threads in
+the same task sequence object.
+
+Example 1 uses the following program:
+
+```
+// A kernel K
+{
+  ...
+  task_sequence<some_function, ...> task_obj1; // Object_1_Thread
+  task_obj1.async(...); // Object_1_Task_1_Thread
+  task_obj1.async(...); // Object_1_Task_2_Thread
+  ...
+  task_sequence<some_function, ...> task_obj2; // Object_2_Thread
+  task_obj2.async(...); // Object_2_Task_1_Thread
+  task_obj2.async(...); // Object_2_Task_2_Thread
+}
+```
+The call to the task_sequence constructor creates the task object threads
+_Object_1_Thread_ and _Object_2_Thread_. The first two `async` calls create task
+threads _Object_1_Task_1_Thread_ and _Object_1_Task_2_Thread_. Similarly the
+next two calls create _Object_2_Task_1_Thread_ and _Object_2_Task_2_Thread_.
+
+The table below provides a view of the hierarchy of task threads that will be
+generated.
+
+.Hierarchy of task threads.
+[cols="s,,,,"]
+|=====
+// row 1, cells 2 spans 4 cells hence the '4+' before '|'
+| Work Item 4+^| _WI_
+// row 2, cells after the first one span 2 cells each
+|Task Sequence Object Thread
+2+^|_Object_1_Thread_
+2+^| _Object_2_Thread_
+// row 3
+| Task Thread
+^| _Task_1_1_
+^|__Task_1_2__
+^|_Task_2_1_
+^|_Task_2_2_
+|=====
+
+At some initial stage, all task threads have weakly parallel forward progress
+guarantee. If _K_ is strengthened to have concurrent forward progress
+guarantee, then all of the object threads are also strengthened. Next, in this
+example one task thread for each task sequence is also strengthened. This is
+depicted in the table below (progress guarantee for each thread is in
+parenthesis):
+
+.Possible Progress Guarantees at some time after _WI_ is strengthened.
+[cols="s,,,,"]
+|=====
+// row 1, cells 2 spans 4 cells hence the '4+' before '|'
+| Work Item
+4+^| _WI_ (concurrently parallel)
+// row 2, cells after the first one span 2 cells each
+|Task Sequence Object Thread
+2+^|_Object_1_Thread_ (concurrent)
+2+^| _Object_2_Thread_ (concurrent)
+// row 3
+| Task Thread
+^| _Task_1_1_ (weakly parallel)
+^|__Task_1_2__ (concurrent)
+^|_Task_2_1_ (concurrent)
+^|_Task_2_2_ (weakly parallel)
+|=====
+
+The next example shows how a task thread delegates its progress
+guarantee to another task thread:
+
+Assume that we have a task sequence object _TS_ templated on the function `f`
+with `response_capacity` of 1 and `invocation_capacity` of 5. Four `async` calls
+create the following task threads: _T1_, _T2_, _T3_ and _T4_, for _TS_. _T1_ has
```

Comment:
I want to emphasize that there's no guarantee that the first asynched thread is the one that will get strengthened.  

```suggestion
create the following task threads: _T1_, _T2_, _T3_ and _T4_, for _TS_. Say _T1_ has
```

Reply:
Is it obvious from earlier descriptions that the order of strengthening of the threads in no way affects the order of retrieving the results?

Reply:
Good catch.  I don't think it is obvious.  See my comment about response_capacity for more details.